AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -212,6 +212,7 @@ - (instancetype)_initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionV
   
   _registeredSupplementaryKinds = [NSMutableSet set];
   
+  _cellsForVisibilityUpdates = [[NSMutableArray alloc] init];
```

Comment:
Let's make this a mutable set.  It will not be quite as efficient, but It will address the possibility of calls being out of sync, and prevent anything from being added twice. This is mainly a risk on IOS seven where there is not a true implementation for the will display cell call, and we call it manually.